### ISSUE TITLE
Clean up VC `secrets-dir` flag and docs

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#5331]
## Description

@chong-he noticed that the VC flag `--secrets-dir` conflicts with `datadir`:

https://github.com/sigp/lighthouse/blob/b5bae6e7a2f52310617a3b93ab7579f3dc9d7880/validator_client/src/cli.rs#L65-L77

This is despite SigP using them together in our infra!

The trick is to put the datadir flag first:

```bash
# This works
lighthouse --datadir $PATH vc --secrets-dir $SECRETS

# This doesn't
lighthouse vc --datadir $PATH --secrets-dir $SECRETS
```

It seems we should _remove_ the `conflicts_with` clause.

Additionally, `secrets-dir` doesn't seem to function as its docs describe:

> The directory which contains the password to unlock the validator \
voting keypairs. Each password should be contained in a file where the \
name is the 0x-prefixed hex representation of the validators voting public \
key. Defaults to ~/.lighthouse/{network}/secrets.

The VC uses the flag for two things:

1. During validator "discovery" of keys that are not already present in the validator definitions YAML, here: https://github.com/sigp/lighthouse/blob/b5bae6e7a2f52310617a3b93ab7579f3dc9d7880/common/account_utils/src/validator_definitions.rs#L326-L330
If a key already has a validator definition, then the VC will not attempt to read the password from the `secrets-dir`, it will expect `voting_keystore_password_path` to have been set accordingly.
2. When creating validators via the HTTP API and the `--http-store-passwords-in-secrets-dir` is used.

The docs make it seem like Lighthouse might try to read the password from the `secrets-dir` if no password or path is present in the definitions, but this is _not the case_.

## Version

v5.0.0

## Steps to resolve

- Remove `conflicts_with`.
- Update the docs to clarify the behaviour.

**Comment by chong-he** [comment]
I would like to work on this

**Comment by michaelsproul** [comment]
Another option:

- Use the `secrets-dir` even when the validator definitions contains an entry for a key

**Comment by chong-he** [comment]
Closed by #5480
